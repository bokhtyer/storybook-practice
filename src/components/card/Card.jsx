export default function Card({ title, content, alignment = "left" }) {
    return (
        <div
            style={{
                border: "1px solid black",
                textAlign: alignment,
                width: "300px",
                padding: "10px",
                borderRadius: "5px",
            }}
            data-testid="card-item"
        >
            <h2>{title}</h2>
            <p>{content}</p>
        </div>
    );
}
