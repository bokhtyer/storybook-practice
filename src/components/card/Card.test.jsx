import { render, screen } from "@testing-library/react";
import Card from "./Card";
import "@testing-library/jest-dom";

test("renders Card with title", () => {
    const { asFragment } = render(<Card title="My Card Title A" content="Hello content" alignment="center" />);
    expect(screen.getByText("My Card Title A")).toBeInTheDocument();
    expect(screen.getByText("Hello content")).toBeInTheDocument();
    expect(screen.getByTestId("card-item")).toHaveStyle("text-align: center");

    expect(asFragment()).toMatchSnapshot();
});
