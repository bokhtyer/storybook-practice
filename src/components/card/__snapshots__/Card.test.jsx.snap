// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card with title 1`] = `
<DocumentFragment>
  <div
    data-testid="card-item"
    style="border: 1px solid black; text-align: center; width: 300px; padding: 10px; border-radius: 5px;"
  >
    <h2>
      My Card Title A
    </h2>
    <p>
      Hello content
    </p>
  </div>
</DocumentFragment>
`;
